src/rabbit_auth_backend_ldap.erl

```diff
@@ -510,11 +508,13 @@ with_login(Creds, Servers, Opts, Fun, RetriesLeft) ->
                                 [scrub_dn(UserDN, env(log))]),
                             {refused, UserDN, []};
                         {error, ldap_closed} ->
+                            purge_connection(Creds, Servers, Opts),
                             with_login(Creds, Servers, Opts, Fun, RetriesLeft - 1);
                         {error, {gen_tcp_error, _}} ->
+                            purge_connection(Creds, Servers, Opts),
                             with_login(Creds, Servers, Opts, Fun, RetriesLeft - 1);
                         {error, E} ->
-                            ?L1("bind error: ~s ~p",
+                            ?L1("bind error: ~p ~p",
```

Comment:
I modified this because the format of `E` doesn't always follow `~s ~p` as seen in the log:

```
2018-05-30 10:36:17.087 [info] <0.291.0> FORMAT ERROR: "    LDAP bind error: ~s ~p~n" [{error,ldap_connect_error},{'EXIT...
```